R/survey_statistics.r

```diff
@@ -516,10 +523,124 @@ survey_median <- function(
     level <- level[1]
   }
 
-  survey_quantile(
+  out = survey_quantile(
     x, quantiles = 0.5, na.rm = na.rm, vartype = vartype, level = level, q_method = q_method,
     f = f, interval_type = interval_type, ties = ties, df = df, .svy = .svy
   )
+  names(out) = sub("^_q50", "", names(out))
+  out
+  # survey_quantile(
+  #   x, quantiles = 0.5, na.rm = na.rm, vartype = vartype, level = level, q_method = q_method,
+  #   f = f, interval_type = interval_type, ties = ties, df = df, .svy = .svy
+  # )
+}
+
+#' Calculate the population variance and its variation using survey methods
+#'
+#' Calculate population variance from complex survey data. A wrapper
+#' around \code{\link[survey]{svyvar}}. \code{survey_var} should always be
+#' called from \code{\link{summarise}}.
+#'
+#' @param x A variable or expression, or empty
+#' @param na.rm A logical value to indicate whether missing values should be dropped
+#' @param vartype Report variability as one or more of: standard error ("se", default)
+#'                or variance ("var") (confidence intervals and coefficient
+#'                of variation not available).
+#' @param .svy A \code{tbl_svy} object. When called from inside a summarize function
+#'             the default automatically sets the survey to the current survey.
+#' @param ... Ignored
+#' @details
+#' Sampling distribution of the variance statistic in general is asymmetric and
+#' if analised variable isn't normally distributed or there is huge variation in
+#' sampling probabilities of the survey design (or both) it could converge to
+#' normality only very slowly (with growing number of survey design degrees of
+#' freedom). Because of that constructing confidence intervals for population
+#' variance is in general a complicated task and can't be done with the same
+#' methods srvyr package provides for other statistics.
+#' @examples
+#' library(survey)
+#' data(api)
+#'
+#' dstrata <- apistrat %>%
+#'   as_survey_design(strata = stype, weights = pw)
+#'
+#' dstrata %>%
+#'   summarise(api99_var = survey_var(api99),
+#'             api99_sd = survey_sd(api99))
+#'
+#' dstrata %>%
+#'   group_by(awards) %>%
+#'   summarise(api00_var = survey_var(api00),
+#'             api00_sd = survey_sd(api00))
+#'
+#' # standard deviation and variance of the population variance estimator
+#' # are available with vartype argument
+#' # (but not for the population standard deviation estimator)
+#' dstrata %>%
+#'   summarise(api99_variance = survey_var(api99, vartype = c("se", "var")))
+#' @export
+
+survey_var <- function(
+  x, na.rm = FALSE, vartype = c("se", "var"), .svy = current_svy(), ...
+) {
+  UseMethod("survey_var", .svy)
+}
+
+#' @export
+survey_var.tbl_svy <- function(
+  x, na.rm = FALSE, vartype = c("se", "var"), .svy = current_svy(), ...
+) {
+  if (!is.null(vartype)) {
+    vartype <- if (missing(vartype)) "se" else match.arg(vartype, several.ok = TRUE)
+  }
+  df <- Inf
```

Comment:
Any reason not to have df as an argument?

Looking back through this code, I see that `svy_mean` and `svy_quantile` are not consistent about the defaults. I prefer the way svy_mean works, where it pulls the degrees of freedom from the object, and will make an issue to change it for svy_quantile.

Reply:
(eg https://github.com/gergness/srvyr/pull/48/files#diff-d35eccba66d9e9af4a631c87f7d76f57R87)

Reply:
Generally that's because I decided not to allow returning confidence intervals for variance statistics (and that is where `df` is used with other statistics). A general way to compute confidence intervals limits provided by `get_var_est()` is not valid for variance, as its sampling distribution is asymptotically chi-squared (irrespective population distribution of a variable) . However convergence for non-normally distributed variables is not very fast even in case of SRS and will be even slower when it comes to complex survey designs. Because of this I decided to allow only "se" and "var" `vartype` in `svy_var()` making use of `df` useless.

Reply:
So finally I made `survey_var` mimics what other functions do regarding CI and df with only one exception: I made `df=Inf` to be default. That's because in case of variance statistic if sampling distribution (lastly) converges to something acceptably symmetric, this will be simply a normal distribution (and if it not converges there is very little reason for using _t_ distribution instead).